app/controllers/application_controller.rb

```diff
@@ -1,7 +1,14 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
-  def current_user
-    User.find(params[:user_id])
+  protect_from_forgery with: :exception
+
+  before_action :update_allowed_parameters, if: :devise_controller?
+
+  protected
+
+  def update_allowed_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password) }
+    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :email, :password, :current_password) }
   end
 end
```

Comment:
## [OPTIONAL]

- [ ] well done your authentication is working properly, however, I would suggest you authenticate users before any action. With this, no action will be permitted for a user who is not logged In. Hint :point_down: 

```rb 
class ApplicationController < ActionController::Base
protect_from_forgery with: :exception
 before_action :authenticate_user!


  before_action :update_allowed_parameters, if: :devise_controller?

  protected

  def update_allowed_parameters
    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:name, :email, :password) }
    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:name, :email, :password, :current_password) }
  end
end
```